src/Function.FromCallback.tt

```diff
@@ -54,52 +53,40 @@ foreach (var (hasCaller, resultCount, parameterCount, methodGenerics, delegateTy
                 throw new ArgumentNullException(nameof(callback));
             }
 
-            var parameterKinds = new List<ValueKind>();
-            var resultKinds = new List<ValueKind>();
-
-<# if (callbackParameterTypeExpressions.Length == 0) { #>
-            var callbackParameterTypes = Array.Empty<Type>();
-<# } else { #>
-            var callbackParameterTypes = new Type[] { <#= callbackParameterTypeExpressions #>};
-<# } #>
-
-            var callbackReturnType = <#= callbackReturnTypeExpression #>;
-
-            using var funcType = Function.GetFunctionType(callbackParameterTypes, callbackReturnType, parameterKinds, resultKinds, allowCaller: <#= hasCaller ? "true" : "false" #>, allowTuple: <#= (resultCount > 1) ? "true" : "false" #>, out _, out _);
-
-            if (funcType is null)
-            {
-                // This means a parameter/result type combination was used that cannot
-                // be represented with the current generic parameters. Therefore, fall
-                // back to using reflection.
-                return FromCallback(store, (Delegate)callback);
-            }
-
+            var (parameterKinds, resultKinds) = GetFunctionType(<# if (callbackParameterTypeExpressions.Length == 0) { #>Array.Empty<Type>()<# } else { #>new Type[] { <#= callbackParameterTypeExpressions #>}<# } #>, <#= callbackReturnTypeExpression #>, allowCaller: <#= hasCaller ? "true" : "false" #>, allowTuple: <#= (resultCount > 1) ? "true" : "false" #>);
 <#= parameterConverters
 #><#= resultConverters
 #>
             unsafe
             {
-                Function.Native.WasmtimeFuncUncheckedCallback func = (env, callerPtr, args_and_results, num_args_and_results) =>
+                Native.WasmtimeFuncUncheckedCallback func = (env, callerPtr, args_and_results, num_args_and_results) =>
                 {
 <# GenerateCallbackContent(hasCaller, resultCount, parameterCount); #>
                 };
 
-                Native.wasmtime_func_new_unchecked(
-                    store.Context.handle,
-                    funcType,
-                    func,
-                    GCHandle.ToIntPtr(GCHandle.Alloc(func)),
-                    Finalizer,
-                    out var externFunc
-                );
+                var funcType = CreateFunctionType(parameterKinds, resultKinds);
+                var externFunc = new ExternFunc();
```

Comment:
Nit: If the `out` modifier for the `Externfunc` parameter in `wasmtime_func_new_unchecked` is correct (which means the function will always write the value), it should suffice to just declare the variable:
```suggestion
                ExternFunc externFunc;
```

Otherwise (if the called function does not always write the value) it should be declared as `ref` parameter instead (but I assume the `out` declaration is correct here).

Reply:
Fixed it here and in one other place.